packages/web-app-files/src/mixins/fileActions.js

```diff
@@ -49,11 +49,11 @@ export default {
           icon: this.apps.meta[editor.app].icon,
           handler: item => this.$_fileActions_openEditor(editor, item.path, item.id),
           isEnabled: ({ resource }) => {
-            if (editor.routes?.length > 0 && !checkRoute(editor.routes, this.$route.name)) {
-              return false
+            if (editor.handler || (editor.routes && checkRoute(editor.routes, this.$route.name))) {
```

Comment:
Logic here is that if `editor.routes` contains elements at all and the current route (`this.$route.name`) is not part of the list, then we interpret the extension as not being available for the current route.  `editor.routes` being empty = extension is available for all routes. Proposing the following:

```
if (editor.handler || editor.routes?.length === 0 || checkRoute(editor.routes, this.$route.name)) {
  return resource.extension === editor.extension
}

return false
```

Reply:
Sure, i've just inverted the check sequencing as it made it clearer to me.  I think this has been changed upstream since the original PR but the important part is just to allow it to be enabled if the `editor.handler` is truthy.

Reply:
@rpocklin how do you suggest to move on, would you mind rebasing this PR? Or should we go ahead and do so?

Reply:
Yep will rebase and update in the next couple of days, thanks.

Reply:
I've updated the PR but changed @kulmann 's recommendation slightly as if `editor.routes = undefined` it would fail.
My understanding is that if `editor.routes = undefined` or `editor.routes = []` it should pass without the third check.  It should only call `checkRoute` if `editor.routes = ['at_least_one_item']`